Assets/_Project/Prototype Pattern/Scripts/Player/PlayerController.cs

```diff
@@ -0,0 +1,80 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace PrototypePattern.Player
+{
+    public class PlayerController : MonoBehaviour, IDamageable
+    {
+        [SerializeField] private float _health;
+        [SerializeField] private bool _targetable = true;
+        [SerializeField] private bool _invencible = false;
+        [SerializeField] private Collider2D _physicsCollider;
+
+        [SerializeField] private Rigidbody2D _rigidBody2D;
+        private InputHandler _inputHandler;
+
+        public float Health
+        {
+            get => _health;
+            set
+            {
+                _health = value;
+                if (_health <= 0)
+                {
+                    Targetable = false;
+                    OnDeath();
+                }
+            }
+        }
+
+        public bool Targetable
+        {
+            get => _targetable;
+            set
+            {
+                _targetable = value;
+            }
+        }
+
+        public bool Invencible
+        {
+            get => _invencible;
+            set
+            {
+                _invencible = value;
+                _physicsCollider.enabled = !Invencible; 
+            }
+        }
+
+        private void Start()
```

Comment:
Prefer populating component references in Awake

```suggestion
        private void Awake()
        {
            _physicsCollider = GetComponent<Collider2D>();
            _rigidBody2D = GetComponent<Rigidbody2D>();
            _inputHandler = GetComponent<InputHandler>();
        }
```

Reply:
I have a question!

I avoid using anything in Awake, because I never know when I am gonna have a Singleton and I don't want to have problem with initialization.

So should I have in my mind to always initialize "GetComponent logic" in Awake? Are there another that I need to keep in my mind?

To be honest, I really don't know what should be in Start or what should be in Awake.